src/snowflake/snowpark/modin/pandas/series.py

```diff
@@ -129,20 +129,64 @@ def __init__(
 
         # modified:
         # Engine.subscribe(_update_engine)
+        from snowflake.snowpark.modin.plugin.extensions.index import Index
```

Comment:
One suggestion for series constructor is that. Could we try us best to reuse dataframe constructor. For example, create a dataframe first from the input and then call `to_series` afterwards.